cheroot/server.py

```diff
@@ -1862,19 +1862,22 @@ def prepare_socket(bind_addr, family, type, proto, nodelay, ssl_adapter):
         """Create and prepare the socket object."""
         sock = socket.socket(family, type, proto)
         prevent_socket_inheritance(sock)
-        if not IS_WINDOWS:
+
+        host, port = bind_addr[:2]
+
+        if not IS_WINDOWS and port != 0:
             # Windows has different semantics for SO_REUSEADDR,
-            # so don't set it.
+            # so don't set it
             # https://msdn.microsoft.com/en-us/library/ms740621(v=vs.85).aspx
+            # Also refer to Issue #114 for why adding SO_REUSEADDR under linux
```

Comment:
Please put the full URL, because repos can move around code hostings over time. You don't know what happens in another decade and what will be used for bug tracking :)